docs: Deploy Loki Helm on AWS guide (#14517)
grafana/loki: docs/sources/setup/install/helm/install-monolithic/_index.md

# Install the monolithic Helm chart

This Helm Chart installation deploys Grafana Loki in [monolithic mode](https://grafana.com/docs/loki/<LOKI_VERSION>/get-started/deployment-modes/#monolithic-mode) within a Kubernetes cluster.

## Prerequisites

- Helm 3 or above. See [Installing Helm](https://helm.sh/docs/intro/install/).
- A running Kubernetes cluster.

## Single Replica or Multiple Replicas

There are two ways to deploy Loki in monolithic mode:
1. **Single Replica**: Run Loki with a single replica. This mode is useful for testing and development or if you are planning to run Loki as a meta-monitoring system.
2. **Multiple Replicas**: Run Loki with multiple replicas. This mode is useful for high availability and scalability. This mode is less economical than distributed mode, but it is simpler to operate.

> Once you have selected choose the appropriate `values.yaml` configuration file below and then continue with the deployment steps.

Reviewer: Once you have selected..... What?  
(Once you have selected how you want to deploy Loki....)

> Once you have selected how many replicas you would like to deploy choose the appropriate `values.yaml` configuration file below and then continue with the deployment steps.

Reviewer: ```suggestion
Once you have selected how many replicas you would like to deploy, choose the appropriate `values.yaml` configuration file below and then continue with the deployment steps.
```

{{< /admonition >}}

```yaml
loki:
  commonConfig:
    replication_factor: 1
  schemaConfig:
    configs:
      - from: 2024-04-01
        store: tsdb
        object_store: s3
        schema: v13
        index:
          prefix: loki_index_
          period: 24h
  pattern_ingester:
      enabled: true
  limits_config:
    allow_structured_metadata: true
    volume_enabled: true
    retention_period: 672h # 28 days retention
  compactor:
    retention_enabled: true 
    delete_request_store: s3
  ruler:
    enable_api: true

minio:
  enabled: true
      
deploymentMode: SingleBinary

singleBinary:
  replicas: 1

# Zero out replica counts of other deployment modes
backend:
  replicas: 0
read:
  replicas: 0
write:
  replicas: 0

ingester:
  replicas: 0
querier:
  replicas: 0
queryFrontend:
  replicas: 0
queryScheduler:
  replicas: 0
distributor:
  replicas: 0
compactor:
  replicas: 0
indexGateway:
  replicas: 0
bloomCompactor:
  replicas: 0
bloomGateway:
  replicas: 0
```

In this configuration, we are deploying Loki with MinIO as the object storage. We recommend configuring object storage via cloud provider or pointing Loki at a MinIO cluster for production deployments.

> ### Multi Replica

Reviewer: ```suggestion
### Multiple Replicas
```

> The default Helm chart deploys the following components:

Reviewer: This line and line 30 can't BOTH be true....  Not if the values are different for single and multiple replicas.  I suspect the difference is based on the value set for `replication_factor`?

> In this configuration, we are deploying Loki with MinIO as the object storage. We recommend configuring object storage via cloud provider or pointing Loki at a MinIO cluster for production deployments.

Author: Same as above comment we can compare minio as equivalent to s3

  replicas: 0
bloomGateway:
  replicas: 0
```
In this configuration, we need to make sure to update the `commonConfig.replication_factor` and `singleBinary.replicas` to the desired number of replicas. We are deploying Loki with MinIO as the object storage. We recommend configuring object storage via cloud provider or pointing Loki at a MinIO cluster for production deployments.

## Deploying the Helm chart for development and testing

1. Add [Grafana's chart repository](https://github.com/grafana/helm-charts) to Helm:

   ```bash
   helm repo add grafana https://grafana.github.io/helm-charts
   ```

1. Update the chart repository:

   ```bash
   helm repo update
   ```

1. Deploy Loki using the configuration file `values.yaml`:

   ```bash
    helm install loki grafana/loki-stack -f values.yaml
    ```
1. Install or upgrade the Loki deployment.
     - To install:
        ```bash
       helm install --values values.yaml loki grafana/loki
       ```
    - To upgrade:
       ```bash
       helm upgrade --values values.yaml loki grafana/loki
       ```
       
1. Verify that Loki is running:
    ```bash
    kubectl get pods -n loki
    ```

## Object Storage Configuration

> After testing Loki with MinIO, we recommend to configure Loki with an object storage provider. The following examples shows how to configure Loki with different object storage providers:

Reviewer: ```suggestion
After testing Loki with MinIO, we recommend configuring Loki with an object storage provider. The following examples shows how to configure Loki with different object storage providers:
```

{{< admonition type="caution" >}}
When deploying Loki using S3 Storage **DO NOT** use the default bucket names;  `chunk`, `ruler` and `admin`. Choose a unique name for each bucket. For more information see the following [security update](https://grafana.com/blog/2024/06/27/grafana-security-update-grafana-loki-and-unintended-data-write-attempts-to-amazon-s3-buckets/). This caution does not apply when you are using MinIO. When using MinIO we recommend using the default bucket names.
{{< /admonition >}}

{{< collapse title="S3" >}}

```yaml
loki:
  commonConfig:
    replication_factor: 3
  schemaConfig:
    configs:
      - from: "2024-04-01"
        store: tsdb
        object_store: s3
        schema: v13
        index:
          prefix: loki_index_
          period: 24h
  storage_config:
    aws:

>       region: <AWS region your bucket is in eg. `eu-west-2`>

Reviewer: ```suggestion
      region: <AWS region your bucket is in, for example, `eu-west-2`>
```

>       bucketnames: <Your AWS bucket for chunk eg. `aws-loki-dev-chunk`>

Reviewer: ```suggestion
      bucketnames: <Your AWS bucket for chunk, for exaxmple,  `aws-loki-dev-chunk`>
```

      s3forcepathstyle: false
  pattern_ingester:
      enabled: true
  limits_config:
    allow_structured_metadata: true
    volume_enabled: true
    retention_period: 672h # 28 days retention

  storage:
    type: s3
    bucketNames:

>         chunks: <Your AWS bucket for chunk eg. `aws-loki-dev-chunk`>
>         ruler: <Your AWS bucket for ruler eg. `aws-loki-dev-ruler`>
>         admin: <Your AWS bucket for admin eg. `aws-loki-dev-admin`>

Reviewer: ```suggestion
        chunks: <Your AWS bucket for chunk, for example, `aws-loki-dev-chunk`>
        ruler: <Your AWS bucket for ruler, for example, `aws-loki-dev-ruler`>
        admin: <Your AWS bucket for admin, for example, `aws-loki-dev-admin`>
```

      useManagedIdentity: false
      # Flag indicating whether to use a federated token for authentication
      useFederatedToken: false
      # Client ID of the user-assigned managed identity (if applicable)
      userAssignedId: <your-user-assigned-id>
      # Timeout duration for requests made to the Azure Blob Storage account (in seconds)
      requestTimeout: <your-request-timeout>
      # Domain suffix of the Azure Blob Storage service endpoint (e.g., core.windows.net)
      endpointSuffix: <your-endpoint-suffix>
    bucketNames:
      chunks: "chunks"
      ruler: "ruler"
      admin: "admin"

# Disable minio storage
minio:
  enabled: false

singleBinary:
  replicas: 3
  persistence:
    storageClass: gp2
    accessModes:
      - ReadWriteOnce
    size: 30Gi

# Zero out replica counts of other deployment modes
backend:
  replicas: 0
read:
  replicas: 0
write:
  replicas: 0

ingester:
  replicas: 0
querier:
  replicas: 0
queryFrontend:
  replicas: 0
queryScheduler:
  replicas: 0
distributor:
  replicas: 0
compactor:
  replicas: 0
indexGateway:
  replicas: 0
bloomCompactor:
  replicas: 0
bloomGateway:
  replicas: 0

```
  
{{< /collapse >}}
  

> To configure other storage providers, refer to the [Helm Chart Reference]({{< relref "../reference" >}}).

Reviewer: ```suggestion
To configure other storage providers, refer to the [Helm Chart Reference](https://grafana.com/docs/loki/<LOKI_VERSION>/setup/install/helm/reference/).
```

## Deploying the Loki Helm chart to a Production Environment

{{< admonition type="note" >}}
We are actively working on providing more guides for deploying Loki in production. 
{{< /admonition >}}

It is recommended to run Loki at scale within in a cloud environment like AWS, Azure, or GCP. The below guides will show you how to deploy a minimally viable production environment.

> - [Deploy Loki on AWS]({{< relref "../deployment-guides/aws" >}})

Reviewer: ```suggestion
- [Deploy Loki on AWS](https://grafana.com/docs/loki/<LOKI_VERSION>/setup/install/helm/deployment-guides/aws)
```
